ui/lib/services/project_service.dart

```diff
@@ -185,7 +193,7 @@ class ProjectService {
     try {
       return await func();
     } on DioError catch (e) {
-      log('Backend communication failed', error: e.error);
+      log('Backend communication failed', error: e.error.toString());
```

Comment:
I don't think this `.toString()` is necessary. (I would expect the logger to do this internally.)